### ISSUE TITLE
FormatException: Could not parse "EAP AI-162.3715353"?

### ISSUE THREAD

**Issue by jy03773322** [flutter/flutter#8373]
log:

Flutter crash report; please file at https://github.com/flutter/flutter/issues.

## command

flutter doctor

## exception

FormatException: Could not parse "EAP AI-162.3715353".

```
package:flutter_tools/src/base/version.dart 48                      Version.Version.parse
package:flutter_tools/src/android/android_studio.dart 64            AndroidStudio.AndroidStudio.fromMacOSBundle
package:flutter_tools/src/android/android_studio.dart 161           AndroidStudio._allMacOS.<fn>
dart:core                                                           Iterable.toList
package:flutter_tools/src/android/android_studio.dart 163           AndroidStudio._allMacOS
package:flutter_tools/src/android/android_studio.dart 124           AndroidStudio.allInstalled
package:flutter_tools/src/android/android_studio_validator.dart 23  AndroidStudioValidator.allValidators
package:flutter_tools/src/doctor.dart 63                            Doctor.validators
package:flutter_tools/src/doctor.dart 121                           Doctor.diagnose
package:flutter_tools/src/commands/doctor.dart 19                   DoctorCommand.runCommand
package:flutter_tools/src/runner/flutter_command.dart 148           FlutterCommand.verifyThenRunCommand
package:flutter_tools/src/runner/flutter_command.dart 119           FlutterCommand.run
package:args/command_runner.dart 194                                CommandRunner.runCommand
package:flutter_tools/src/runner/flutter_command_runner.dart 221    FlutterCommandRunner.runCommand
package:args/command_runner.dart 109                                CommandRunner.run.<fn>
dart:async                                                          Future.Future.sync
package:args/command_runner.dart 109                                CommandRunner.run
package:flutter_tools/src/runner/flutter_command_runner.dart 150    FlutterCommandRunner.run
package:flutter_tools/executable.dart 128                           main.<fn>.<fn>
package:stack_trace                                                 Chain.capture
package:flutter_tools/executable.dart 127                           main.<fn>
package:flutter_tools/src/base/context.dart 76                      AppContext._run
package:flutter_tools/src/base/context.dart 66                      AppContext.runInZone.<fn>
dart:async                                                          runZoned
package:flutter_tools/src/base/context.dart 65                      AppContext.runInZone
package:flutter_tools/executable.dart 98                            main
../packages/flutter_tools/bin/flutter_tools.dart 8                  main
===== asynchronous gap ===========================
dart:async                                                          _Completer.completeError
package:flutter_tools/src/doctor.dart 149                           Doctor.diagnose
===== asynchronous gap ===========================
dart:async                                                          Future.Future.microtask
package:flutter_tools/src/doctor.dart                               Doctor.diagnose
package:flutter_tools/src/commands/doctor.dart 19                   DoctorCommand.runCommand
===== asynchronous gap ===========================
dart:async                                                          Future.Future.microtask
package:flutter_tools/src/commands/doctor.dart                      DoctorCommand.runCommand
package:flutter_tools/src/runner/flutter_command.dart 148           FlutterCommand.verifyThenRunCommand
===== asynchronous gap ===========================
dart:async                                                          _asyncThenWrapperHelper
package:flutter_tools/src/runner/flutter_command.dart               FlutterCommand.verifyThenRunCommand
package:flutter_tools/src/runner/flutter_command.dart 119           FlutterCommand.run
package:args/command_runner.dart 194                                CommandRunner.runCommand
===== asynchronous gap ===========================
dart:async                                                          Future.Future.microtask
package:args/command_runner.dart                                    CommandRunner.runCommand
package:flutter_tools/src/runner/flutter_command_runner.dart 221    FlutterCommandRunner.runCommand
===== asynchronous gap ===========================
dart:async                                                          _asyncThenWrapperHelper
package:flutter_tools/src/runner/flutter_command_runner.dart        FlutterCommandRunner.runCommand
package:args/command_runner.dart 109                                CommandRunner.run.<fn>
dart:async                                                          Future.Future.sync
package:args/command_runner.dart 109                                CommandRunner.run
package:flutter_tools/src/runner/flutter_command_runner.dart 150    FlutterCommandRunner.run
package:flutter_tools/executable.dart 128                           main.<fn>.<fn>
===== asynchronous gap ===========================
dart:async                                                          Future.Future.microtask
package:flutter_tools/executable.dart                               main.<fn>.<fn>
package:stack_trace                                                 Chain.capture
package:flutter_tools/executable.dart 127                           main.<fn>
```

## flutter doctor

```
[✓] Flutter (on Mac OS, channel master)
    • Flutter at /Users/zhai/Documents/Flutter/flutter
    • Framework revision 2a05cfdf07 (2 hours ago), 2017-02-23 16:27:30
    • Engine revision ab09530927
    • Tools Dart version 1.23.0-dev.0.0

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.2)
    • Android SDK at /Users/zhai/Documents/WorkSpace/adt-bundle-mac-x86_64-20130917/sdk
    • Platform android-25, build-tools 25.0.2
    • Java Development Kit (JDK) found: javac 1.8.0_92

[✓] iOS toolchain - develop for iOS devices (Xcode 8.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.2.1, Build version 8C1002
    • ios-deploy 1.9.0
```

**Comment by eseidelGoogle** [comment]
Related to https://github.com/flutter/flutter/issues/8353.

**Comment by jakobr-google** [comment]
Fixed in #8389.
